File: TuyenHoa_2324/Cau2/KTDD.py
def main():
    # Đọc dữ liệu từ file KTDD.INP
    with open("KTDD.INP", "r") as infile:
        st = infile.read().strip()

    # Tạo tập hợp ký tự đại diện
    # Sử dụng set để tự động loại bỏ các ký tự trùng lặp
    representative_set = set(st)

    # Chuyển tập hợp về danh sách và sắp xếp nếu cần
    representative_list = sorted(representative_set)

    # Ghi kết quả vào file KTDD.OUT
    with open("KTDD.OUT", "w") as outfile:
        # Nối các ký tự trong danh sách bằng khoảng trắng
        outfile.write(" ".join(representative_list) + "\n")

if __name__ == "__main__":
    main()

File: THCS/2023_2024/TuyenHoa_2324/Cau2/KTDD.cpp
#include <iostream>
#include <fstream>
#include <set>
#include <string>
#include <algorithm>

int main() {
    // Đọc dữ liệu từ file KTDD.INP
    std::ifstream infile("KTDD.INP");
    std::string st;
    std::getline(infile, st);
    infile.close();

    // Sử dụng set để tự động loại bỏ các ký tự trùng lặp
    std::set<char> representative_set(st.begin(), st.end());

    // Ghi kết quả vào file KTDD.OUT
    std::ofstream outfile("KTDD.OUT");
    for (auto it = representative_set.begin(); it != representative_set.end(); ++it) {
        if (it != representative_set.begin()) {
            outfile << " "; // Thêm khoảng trắng giữa các ký tự
        }
        outfile << *it; // Ghi ký tự vào file
    }
    outfile << std::endl;
    outfile.close();

    return 0;
}
